src/drivers/Kconfig

```diff
@@ -8,3 +8,27 @@ config DMIC
 	  Select for DMIC driver
 
 endmenu
+
+choice CAVS_VERSION
+	depends on CAVS
+	default CAVS_VER_1_5 if APOLLOLAKE
```

Comment:
I'd go for full name `VERSION` everywhere instead of `VER`

Reply:
Wouldn't it be better to use `select` from platform configs instead of list of default from here?
When you add new platform and it has set of predefined configs it would be better to just have list of selects than look for every config that has platform-dependent defaults (however I'm not 100% sure that select works with choices, so please check it ;p).

Reply:
It should really be
```
choice APOLLOLAKE
        depends on ARCH_XTENSA
        select CAVS_VERSION_1_5
        help
            Please say Y here if building for Intel Apollolake platforms......
```

Reply:
@lrgirdwo 
Do you mean amend src/platform/Kconfig as @jajanusz suggested?
'''
config APOLLOLAKE
	bool "Build for Apollolake"
	select BOOT_LOADER
	select IRQ_MAP
...
'''

Reply:
@lgirdwood @bkokoszx 
Apparently you cannot select choices, I recommend you to read this (written by python Kconfig lib author):
https://github.com/zephyrproject-rtos/zephyr/issues/6948#issuecomment-379082370

Usually selects are discouraged, depends/defaults are always better but my only concern is that I want to not have situation where we have `if PLATFORM`s everywhere in the project.

Reply:
select are typically used to model hard-dependencies (e.g. AppoloLake is based on cAVS1.5 and relies on Xtensa).
You want to expose top-level platforms as configurations, and let all the details be set with select.
I've spend nearly half-a-year fixing Linux Kconfigs and I suggest you limit the use of depends - it's a nightmare once the configs become a bit hairy, while the select mechanism is a clearer model.

Reply:
@plbossart @lgirdwood 
I think 'bit hairy' with selects is where nightmare starts. Selects are good when you have 1/2 levels of dependency, with more you are likely to end up with invalid configs, unless you force some simple practices on developers. Please consider this example: https://docs.zephyrproject.org/latest/tools/kconfig/index.html#select-pitfalls